File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.go
func longestConsecutive(nums []int) int {
    mp := make(map[int]bool)
    for _,val:= range nums{
        mp[val]= true
    }
    ans := 0
    for key,_ := range mp{
        ok := mp[key-1]
        if ok==false{
            count,curr := 0,key
            for mp[curr]==true{
                count++
                curr++
            }
            ans=max(ans,count)
        }
    }
    return ans
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.size();
        unordered_map<char,bool> mp;
        int ans = 0;
        int i=0, j=0;
        while(i < n){
            if(mp[s[i]] == false){
                mp[s[i]] = true;
                ans = max(ans, i-j+1);
                i++;
                continue;
            }
            
            while(j<=i && s[j]!=s[i]){
                mp[s[j]] = false;
                j++;
            }
            mp[s[j]] = false;
            j++;
        }

        return ans;
    }
};
